src/Containers/ContributionForm/ContributionForm.js

```diff
@@ -10,23 +10,48 @@ function ContributionForm() {
   const [lastName, setLastName] = useState('')
   const [specialties, setSpecialties] = useState([])
   const [insurances, setInsurances] = useState([])
+  const [typedInsurances, setTypedInsurances] = useState('')
+  const [modifiedInsurances, setModifiedInsurances] = useState([])
   const [cost, setCost] = useState('')
   const [profession, setProfession] = useState('')
   const [street, setStreet] = useState('')
   const [unit, setUnit] = useState('')
   const [city, setCity] = useState('')
   const [zip, setZip] = useState('')
   const [phone, setPhone] = useState('')
+  const [submitionMessage, setSubmitionMessage] = useState('')
   const [error, setError] = useState('')
+  useEffect((modifiedInsurances) => {
+    let mounted = false
+    if (mounted || state) {
+      makePostRequest()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  },[modifiedInsurances])
```

Comment:
Love this requirement to make sure useEffect is only run if modifiedInsurances changes. That was new for me